### PR TITLE
🌱 Improve error message about missing bundles

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Follow up for https://github.com/operator-framework/operator-controller/pull/498#discussion_r1381696321

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/resolution/variablesources/bundle.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ func MakeBundleVariables(
 		// get bundle dependencies
 		dependencies, err := filterBundleDependencies(allBundles, head)
 		if err != nil {
-			return nil, fmt.Errorf("could not determine dependencies for bundle with id %q: %w", id, err)
+			return nil, fmt.Errorf("could not determine dependencies for bundle %q from package %q from catalog %q: %s", head.Name, head.Package, head.CatalogName, err)
```
Comment (m1kola) [inline internal/resolution/variablesources/bundle.go:98]: I don't like the wording: doesn't feel natural. Any suggestions?

Comment (everettraven) [reply internal/resolution/variablesources/bundle.go:98]: A couple options I could think of:
- `could not determine dependencies for bundle %q from package %q in catalog %q`. This is only a slight variation but prevents the repetition of the word "from"
- Something that includes additional context after the error message like: `could not determine dependencies for bundle %q. catalog: %q, package: %q`

Comment (m1kola) [reply internal/resolution/variablesources/bundle.go]: Simply replacing "from" with "in" sounds much better, thanks! I updated the PR with this version.

File: internal/resolution/variablesources/bundle.go [hunk 2]
```diff
@@ -119,7 +120,7 @@ func filterBundleDependencies(allBundles []*catalogmetadata.Bundle, bundle *cata
 			catalogfilter.InBlangSemverRange(requiredPackage.SemverRange),
 		))
 		if len(packageDependencyBundles) == 0 {
-			return nil, fmt.Errorf("could not find package dependencies for bundle %q", bundle.Name)
+			return nil, errors.New("could not find package dependencies for bundle")
```
Comment (joelanford) [inline internal/resolution/variablesources/bundle.go:123]: Can we add more context to this as well? For example:
```suggestion
			return nil, fmt.Errorf("no bundles found matching required package %q in range %q", requiredPackage.PackageName, requiredPackage.SemverRange)
```

**Review by everettraven — APPROVED** [review]
LGTM

**Pull request merged**